clientlibs/js/src/ApiService/ApiService.ts

```diff
@@ -0,0 +1,318 @@
+import { UpGradeClientEnums, UpGradeClientInterfaces, UpGradeClientRequests } from '../types';
+import { DefaultHttpClient } from '../DefaultHttpClient/DefaultHttpClient';
+import {
+  CaliperEnvelope,
+  IExperimentAssignmentv5,
+  IFeatureFlag,
+  IFlagVariation,
+  IGroupMetric,
+  ILogInput,
+  ISingleMetric,
+} from 'upgrade_types';
+import { DataService } from 'DataService/DataService';
+import { IApiServiceRequestParams, IEndpoints } from './ApiService.types';
+import { IMarkDecisionPointParams } from 'UpGradeClient/UpGradeClient.types';
+import * as uuid from 'uuid';
+
+// this variable is used by webpack to replace the value of USE_CUSTOM_HTTP_CLIENT with true or false to create two different builds
+declare const USE_CUSTOM_HTTP_CLIENT: boolean;
+declare const IS_BROWSER: boolean;
+
+export default class ApiService {
+  private context: string;
+  private hostUrl: string;
+  private userId: string;
+  private token: string;
+  private apiVersion: string;
+  private clientSessionId: string;
+  private httpClient: UpGradeClientInterfaces.IHttpClientWrapper;
+  private api: IEndpoints;
+
+  constructor(config: UpGradeClientInterfaces.IConfig, private dataService: DataService) {
+    this.context = config.context;
+    this.hostUrl = config.hostURL;
+    this.token = config.token;
+    this.clientSessionId = config.clientSessionId;
```

Comment:
It might be good idea to create your own sessionId in case user doesn't provide.